module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,59 @@
+#include "compression.hpp"
+#include <algorithm>
+#include <iostream>
+#include <iomanip>
+
+void printMap (std::array<std::array<uint8_t, width>, height> bitmap)
+{
+    for(const auto& line : bitmap)
+    {
+        for (const auto c : line)
+        {
+            std::cout<<std::setw(3)<<(int)c<<' ';
+
+        }
+        std::cout<<'\n';
+    }
+}
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale (std::array<std::array<uint8_t, width>, height> bitmap)
```

Comment:
Maybe type aliases will be good idea here :)

Reply:
I agree you should consider aliases for longer types like std::vector<std::pair<uint8_t, uint8_t>>

Reply:
I do not like aliasing at all. I keep forgetting what is behind alias.
I changed to get some practice.